### PR TITLE
Ci: migrate utils tests to GitHub

### PR DESCRIPTION
**Pull request by karliatto**
## Description

Migrate utils test from GitLab to GitHub with new name `misc` so it is not confused with utils package.

### CODE REVIEW

File: .github/workflows/test-utils.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+name: "[test] utils"
```
Comment (mroz22) [inline .github/workflows/test-utils.yml]: maybe `[test] misc` so that it won't get confused with utils package?

Comment (karliatto) [reply .github/workflows/test-utils.yml]: Yes, makes sense. 7074867e645225c67728114e1b707ae9bb21f992

**Review by mroz22 — APPROVED** [review]

File: .github/workflows/test-misc.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      - run: ./packages/connect/e2e/test-yarn-install.sh
+
+  test-unit:
+    runs-on: ubuntu-latest
+    if: github.repository == 'trezor/trezor-suite'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
```
Comment (mroz22) [inline .github/workflows/test-misc.yml]: does test unit require submodules?

Comment (karliatto) [reply .github/workflows/test-misc.yml]: You are right, they are not needed. Thanks! :+1:

**Pull request merged**